src/test/java/com/squareup/javapoet/TypeSpecTest.java

```diff
@@ -441,6 +441,68 @@ private boolean isJava8() {
         + "}\n");
   }
 
+  /** https://github.com/square/javapoet/issues/253 */
+  @Test public void enumWithAnnotatedValues() throws Exception {
+    TypeSpec roshambo = TypeSpec.enumBuilder("Roshambo")
+        .addModifiers(Modifier.PUBLIC)
+        .addEnumConstant("ROCK", TypeSpec.anonymousClassBuilder("")
+            .addJavadoc("Avalanche!\n")
+            .addAnnotation(ClassName.get("javax.annotation", "Nullable"))
+            .build())
+        .addEnumConstant("PAPER", TypeSpec.anonymousClassBuilder("$S", "flat")
```

Comment:
You can simplify this to just `addEnumConstant("PAPER")`. The rest is distracting because it's unrelated to annotations & enums.